server/app/auth/oidc/OidcCiviFormProfileAdapter.java

```diff
@@ -73,7 +77,13 @@ protected Optional<String> getAuthorityId(OidcProfile oidcProfile) {
 
   /** Merge the two provided profiles into a new CiviFormProfileData. */
   public CiviFormProfileData mergeCiviFormProfile(
```

Comment:
I don't believe this parent method is ever called with an empty optional correct?

 so another "smell" that likely implies we need more refactoring subsequently.   This may need to be a protected method with a different name for instance.

Reply:
The parent method is called with an empty optional if there's no existing profile. This replaces the existing logic in `createCiviFormProfileFromOidcProfile`, combining the two methods with similar logic into one.

Reply:
To rephrase, the only callers of this OIDC Adapter mergeCiviFormProfile, are the children's [overrides](https://github.com/seattle-uat/civiform/search?q=mergeCiviFormProfile) as far as I see.

And they all ensure a non empty Optional is passed in here, which means that handling that case is not necessary here. 

So the fact that we don't need it but must have it due to inheritance likely indicates an abstraction issue.  Given it's not actually called outside of its children, indicates that this functionality doesn't need to be inherited and is more of a utility to the children.

A slightly better definition might be

```
/**
 * Some doc indicating that if the optional is empty one will be made.
 */
abstract public CiviFormProfileData mergeCiviFormProfile(Optional<CiviFormProfile> maybeCiviformProfile, OidcProfile oidcProfile);

/** Merge the two provided profiles into a new CiviFormProfileData. */
protected CiviFormProfileData mergeCiviFormProfile(CiviFormProfile civiformProfile, OidcProfile oidcProfile) {
...... existing code
}
```

Reply:
The parent's `mergeCiviFormProfile` is passed into the `CiviFormProfileMerger` on line 137. It then gets called inside CiviFormProfileMerger. This behavior is not overridden by the children.

Reply:
This seems good. The parent takes the Optional and resolves it rather than all the children.

Can the non-optional method on L94 be private now?

Reply:
It can't be private because IdcsProfileAdapter overrides it.

Reply:
ah ok, thanks.

Yeah add the VisibleForTesting.